planex/cmd/createmockconfig.py

```diff
@@ -0,0 +1,123 @@
+"""
+planex-create-mock-config: Populate mock config with yum repositories
+"""
+
+from __future__ import print_function
+import argparse
+import os
+import os.path
+import pprint
+import sys
+import yum
+
+import ConfigParser
+import StringIO
+
+import argcomplete
+
+from planex.util import add_common_parser_options
+from planex.util import setup_logging
+from planex.util import setup_sigint_handler
+
+
+def load_mock_reference(fname):
+    """
+    read in the reference mock configuration
+    """
+    config_opts = {}
+    execfile(fname)
+    return config_opts
+
+
+def load_yum_repos(repo_names):
+    """
+    read in the yum repository configuration
+    """
+    yum_base = yum.YumBase()
+    yum_repos = []
+    for repo_id in repo_names:
+        yum_repos += yum_base.repos.findRepos(repo_id)
+    return yum_repos
+
+
+def update_mock_repos(config, yum_repos):
+    """
+    Replace all repository sections with a new one derived from yum
+    """
+    for section in [s for s in config.sections() if s != 'main']:
+        config.remove_section(section)
+    for repo in yum_repos:
+        config.add_section(repo.id)
+        config.set(repo.id, 'name', repo.name)
+        if len(repo.baseurl):
+            config.set(repo.id, 'baseurl', ' '.join(repo.baseurl))
+        else:
+            config.set(repo.id, 'mirrorlist', repo.mirrorlist)
+        if repo.gpgcheck:
+            config.set(repo.id, 'gpgcheck', '1')
+            config.set(repo.id, 'gpgkey', ' '.join(repo.gpgkey))
+        else:
+            config.set(repo.id, 'gpgcheck', '0')
+
+
+def write_mock_cfg(fileh, cfg):
+    """
+    Save the mock configuration to a file handle
+    """
+    for key, value in cfg.items():
+        if key == 'yum.conf':
+            print('config_opts[\'%s\'] = """\n%s"""' % (key, value),
+                  file=fileh)
+        else:
```

Comment:
This could be confused by a mock config with a `dnf.conf` section.   I think the `dnf.conf` format is the same as yum.conf: if so, then the 'yum.conf' branch should also handle `dnf.conf`; if not, you should probably bail out if you see `dnf.conf`.